### ISSUE TITLE
graph/encoding/dot: Edge chains do not retain their attributes

### ISSUE THREAD

**Issue by wesalvaro** [gonum/gonum#545]
### What are you trying to do?

I am trying to encode a DOT file.

### What did you do?

```dot
A -> B -> C [len=5]
```

### What did you expect to happen?

The above should be interpreted as:

```dot
A -> B [len=5]
B -> C [len=5]
```

### What actually happened?

The above was interpreted as:

```dot
A -> B [len=5]
B -> C
```

### What version of Go and Gonum are you using?

go version go1.10.3 darwin/amd64
ff99a9821f0fc7e0d40749b5148e54a57ea97d6e

### Does this issue reproduce with the current master?

Yes.

**Comment by wesalvaro** [comment]
The following test change demonstrates the issue:

```diff
diff --git a/graph/encoding/dot/decode_test.go b/graph/encoding/dot/decode_test.go
index 13f4002a..af07dae6 100644
--- a/graph/encoding/dot/decode_test.go
+++ b/graph/encoding/dot/decode_test.go
@@ -86,7 +86,7 @@ const directed = `digraph {
        B [label="bar 2"];

        // Edge definitions.
-       A -> B [label="baz 2"];
+       A -> B -> A [label="baz 2"];
 }`

 const undirected = `graph {
```

**Comment by kortschak** [comment]
@mewmew, do you have time to look into this?

**Comment by mewmew** [comment]
@kortschak PR submitted #546

@wesalvaro Thanks for reporting this issue. It should have been fixed in #546. If you'd like to contribute, would you try adding a test case for it? That would be much appreciated :)

(I'm on travelling foot at the moment so only have momentarily access to the Internet.)

Cheers,
/u

**Comment by kortschak** [comment]
I have a test. I'll send the PR after this is merged - since you're walking I'll do that too.
